### PR TITLE
test-fixes

### PR DESCRIPTION
**Pull request by Connorbelez**
.gitnexus

.gitnexus

fix: address crm origination and rotessa regressions

chore: apply additional test and admin/origination fixes

## Summary by Sourcery

Refine payment obligation cron processing and monitoring, harden Rotessa PAD webhook handling, and adjust origination, CRM, and auth flows to fix regressions while tightening tests and access controls.

New Features:
- Introduce a reusable dependency-injected implementation for the obligation transition cron to enable fine-grained simulation and monitoring.
- Add a query to resolve Rotessa webhook transfer context so provider-managed occurrences are only ingested when appropriate.
- Expose an origination collections activation runtime helper for reuse across admin workflows.

Bug Fixes:
- Correct grace period calculations for obligations to use a seven-day window consistently across generation and cron tests.
- Fix obligation transition cron overflow tracking so zero-activity business days are recorded and wave-based metrics and alerts behave correctly.
- Ensure Rotessa PAD webhooks fall back to the direct transfer path when no provider-managed schedule linkage exists, preventing incorrect ingestion attempts.
- Preserve payment bootstrap-derived IDs when reading existing mortgage activation results instead of re-querying obligations and plan entries.
- Prevent FairLend admin role from implicitly granting all permissions without the explicit admin:access permission.
- Require explicit org context checks for origination case access across queries and mutations, including CRM listing detail contexts.
- Handle provider-managed immediate collections activation more robustly by normalizing draft state and separating runtime logic from the action wrapper.
- Avoid misrouting workout plan mutations by switching to authenticated mutations for create, activate, complete, and cancel operations.
- Prevent Rotessa provider-managed occurrence ingestion from running for unrelated transfers and add logging for direct-transfer webhooks.
- Fix AMPS demo payout idempotency expectations and CRM view column visibility behavior when users override column visibility.

Enhancements:
- Refactor the obligation transition cron into a dependency-injected core with structured logging, overflow metrics recording, and alerting, with a local simulation harness for tests.
- Extend origination collections drafts to track provider-managed activation status in sync with the main activation status.
- Reuse the collections activation runtime helper from commit and retry flows to simplify activation orchestration.
- Tighten CRM view-state column visibility semantics so saved views override default visibility instead of merging with it.
- Improve CRM detail queries to validate listing and mortgage access more clearly before returning context.
- Add a helper for building stable participant field IDs and reuse it in the origination participants step UI.
- Clarify admin record sidebar behavior to show loading placeholders and unavailable states instead of silently skipping adapter details.
- Filter demo AMPS ledger account lookups by the correct mortgage identifier, including simulations.
- Adjust EFT VoPay webhook tests to consume shared cash ledger test utilities for consistency.
- Broaden the ledger regression test to allow limited, validated mutations file changes while enforcing required helper extraction patterns.

Tests:
- Add a standalone simulation harness and extensive tests for the obligation transitions cron covering failures, overflow streaks, wave draining, and alert behavior.
- Update payments generation tests to reflect the new seven-day grace period and cleanup obligation tests.
- Extend auth policy tests to assert that FairLend admin without admin:access does not implicitly gain ledger or payment permissions.
- Add CRM detail context tests and admin UI tests for record sidebar and origination participants/collections behavior.
- Enhance ledger regression verification tests to permit vetted mutation changes while guarding critical handler structure.
- Adjust webhook and payout tests (Rotessa, VoPay, reversal, and payout flows) to validate the refined behavior and idempotency semantics.

Chores:
- Remove committed coverage artifacts from the repository and rely on generated coverage instead.

## Summary by CodeRabbit

* **New Features**
  * Added context-aware authorization validation for origination case access with improved error messaging.
  * Introduced webhook transfer context query for enhanced payment processing.

* **Bug Fixes**
  * Permission checks now require explicit admin access permission rather than role-based bypass.
  * Fixed payout replay idempotency behavior to prevent duplicate payouts.
  * Improved view state column visibility handling with explicit override flag.

* **Tests**
  * Refactored obligation processing tests with new simulation harness.
  * Updated multiple test utilities and import paths for consistency.

* **Chores**
  * Removed auto-generated coverage artifacts.

### CODE REVIEW

**Comment by Connorbelez** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.com/github/pr/Connorbelez/tanstackTemplate/427?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#427** <a href="https://app.graphite.com/github/pr/Connorbelez/tanstackTemplate/427?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/Connorbelez/tanstackTemplate/427?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#416** <a href="https://app.graphite.com/github/pr/Connorbelez/tanstackTemplate/416?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 4 other dependent PRs ([#417](https://github.com/Connorbelez/tanstackTemplate/pull/417) <a href="https://app.graphite.com/github/pr/Connorbelez/tanstackTemplate/417?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#424](https://github.com/Connorbelez/tanstackTemplate/pull/424) <a href="https://app.graphite.com/github/pr/Connorbelez/tanstackTemplate/424?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#425](https://github.com/Connorbelez/tanstackTemplate/pull/425) <a href="https://app.graphite.com/github/pr/Connorbelez/tanstackTemplate/425?utm_source=stack-comment-icon" target="_blank">Graphite</a> and 1 other)
* **#415** <a href="https://app.graphite.com/github/pr/Connorbelez/tanstackTemplate/415?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#413** <a href="https://app.graphite.com/github/pr/Connorbelez/tanstackTemplate/413?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#412** <a href="https://app.graphite.com/github/pr/Connorbelez/tanstackTemplate/412?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#411** <a href="https://app.graphite.com/github/pr/Connorbelez/tanstackTemplate/411?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#410** <a href="https://app.graphite.com/github/pr/Connorbelez/tanstackTemplate/410?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#408** <a href="https://app.graphite.com/github/pr/Connorbelez/tanstackTemplate/408?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#407** <a href="https://app.graphite.com/github/pr/Connorbelez/tanstackTemplate/407?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: convex/payments/__tests__/crons.test.ts [hunk 1]
```diff
@@ -264,284 +345,119 @@ describe("processObligationTransitions", () => {
...
+				now: Date.parse(`2026-03-${day}T06:00:00.000Z`),
+			});
 		}
+
+		const monitoring = harness.getMonitoring();
+		expect(monitoring?.newlyDueOverflowStreak).toBe(0);
+		expect(harness.getErrors()).toHaveLength(0);
 	});
 
 	it("preserves legacy obligation date fields as numeric timestamps", async () => {
```
[the remark this answers is not in the document]
Comment (Connorbelez) [reply convex/payments/__tests__/crons.test.ts]: **Fix:** Added  invocation via  to the test so it actually exercises the cron path and verifies // remain numeric after processing, rather than only asserting on seeded data.

```ts
// Before: only validated seeded data
const obligation = await t.run(async (ctx) => ctx.db.get(obligationId));
expect(typeof obligation?.dueDate).toBe(number);

// After: invokes cron then validates post-processing state
await t.action(
  internal.payments.obligations.crons.processObligationTransitions,
  {}
);
const obligation = await t.run(async (ctx) => ctx.db.get(obligationId));
expect(typeof obligation?.dueDate).toBe(number);
```
